File: apps/backend/TecoApi/Models/Entities/Requester.cs
namespace TecoApi.Models.Entities;

public class Requester
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public required User User { get; set; }

    public ICollection<Order> Orders { get; set; } = [];
    public ICollection<Request> Requests { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];

}

File: apps/web/app/types/requester.ts
import User from "@/types/user";
import Request from "@/types/request";
import Order from "@/types/order";
import Review from "@/types/review";

interface Requester {
    id: number;
    userId: number;
    user?: User;
    orders: Order[];
    requests: Request[];
    reviews: Review[];
}

export default Requester;